Automattic/pocket-casts-android
# Display loading progress bar when is filtering Podcasts by Category

## Description
- Display the existing progress bar when is filtering a category or when is loading the category list
- Update the search background color
- I also made a minor refactor: 61ecd428679dbb1d615c97a56d21d88f19f4815d and ac235f968c785c879181d15d4a33f81b5b493b27 which was only renames


## Testing Instructions
- Have the new discover redesign enabled in Beta features
- Sometimes it's difficult to notice the progress bar because it finishes loading quickly, but you should see the Discover progress bar when you tap on any category, especially if it takes more time to fetch data from the API.


## Screenshots or Screencast 
[Screen_recording_20240326_092149.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/ab22b263-63b6-428a-834e-4cdd90b2085a)


## Checklist
 
#### I have tested any UI changes...
